mlir/lib/Dialect/MIOpen/Generator/Conv2dGenerator.cpp

```diff
@@ -118,6 +117,31 @@ Type Conv2dGenerator::getDataType(OpBuilder &builder) const {
   return dataType;
 }
 
+Optional<miopen::ConvOpType>
+Conv2dGenerator::getOpForName(const StringRef name) {
+  if (name == "conv2d") {
+    return miopen::Conv2DOpType;
+  }
+  if (name == "conv2d_bwd_data") {
+    return miopen::Conv2DBwdDataOpType;
+  }
+  if (name == "conv2d_bwd_weight") {
+    return miopen::Conv2DBwdWeightOpType;
+  }
+  return llvm::None;
+}
+
+const char *Conv2dGenerator::getNameForOp(const miopen::ConvOpType op) {
+  switch (op) {
+  case miopen::Conv2DOpType:
+    return "conv2d";
+  case miopen::Conv2DBwdDataOpType:
+    return "conv2d_bwd_data";
+  case miopen::Conv2DBwdWeightOpType:
+    return "conv2d_bwd_weight";
+  }
```

Comment:
Extendibility-wise, it is probably easier if you handle it the same way: use optional such that this reminds a developer certain situation needs to be handled when a new operation type is added, for example.

Reply:
Well, unlike the string->enum function, this one is total, so there's no point in returning None - it'll never happen. And if someone adds a new case to the enum, they'll get compiler warnings about this one.